File: datasets/IR-Plag/convertai-original-java-rest-kotlin/case-07/plagiarized/L4/09/Level4.kt
package IR

import java.util.Scanner

/**
 * @author FD8DBE9073AFCC0504CD6901E1091CAD
 */
fun main() {
    val `in` = Scanner(System.`in`)
    val matriks = Array(4) { DoubleArray(4) }
    var sum = 0.0

    print("Enter a 4 by 4 matrix row by row: ")
    for (b in 0 until 4) {
        for (k in 0 until 4) {
            matriks[b][k] = `in`.nextDouble()
        }
    }

    for (x in matriks.indices) {
        sum += matriks[x][x]
    }

    println("Sum of the elements in the major diagonal is $sum")
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L4/09/Level4.ts
import { Scanner } from 'readline-sync';

const in: Scanner = new Scanner();

let matriks: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));
let sum: number = 0;

console.log("Enter a 4 by 4 matrix row by row: ");
for (let b: number = 0; b < 4; b++) {
    for (let k: number = 0; k < 4; k++) {
        matriks[b][k] = in.nextDouble();
    }
}

for (let x: number = 0; x < matriks.length; x++) {
    sum += matriks[x][x];
}

console.log(`Sum of the elements in the major diagonal is ${sum}`);

